vsphere/resource_vsphere_compute_cluster.go

```diff
@@ -1518,6 +1649,57 @@ func resourceVSphereComputeClusterApplyVsanConfig(d *schema.ResourceData, meta i
 		}
 	}
 
+	// handle file service
+	if d.Get("vsan_file_service_enabled").(bool) && d.HasChange("vsan_file_service_enabled") {
+		fileServiceOvfUrl, err := vsanclient.FindOvfDownloadUrl(meta.(*Client).vsanClient, cluster.Reference())
+		if err != nil {
+			return fmt.Errorf("cannot find vsan file service OVF url, err: %s", err)
+		}
+
+		if err := vsanclient.DownloadFileServiceOvf(meta.(*Client).vsanClient, meta.(*Client).vimClient, fileServiceOvfUrl); err != nil {
+			return fmt.Errorf("cannot download vsan file service OVF with url: %s", fileServiceOvfUrl)
+		} else {
+			log.Printf("[DEBUG] downloaded vsan file service OVF with url: %s", fileServiceOvfUrl)
+		}
+
+		fileServiceConf := d.Get("vsan_file_service_conf").(*schema.Set).List()[0].(map[string]interface{})
+
+		fileServiceConfig, err := expandVsanFileServiceConfig(d, meta, fileServiceConf)
+		if err != nil {
+			return err
+		}
+		if err := vsanclient.Reconfigure(meta.(*Client).vsanClient, cluster.Reference(), vsantypes.VimVsanReconfigSpec{
+			Modify:            true,
+			FileServiceConfig: fileServiceConfig,
+		}); err != nil {
+			return fmt.Errorf("cannot apply vsan file service on cluster '%s': %s", d.Get("name").(string), err)
+		}
+
+		if len(fileServiceConf["vsan_file_service_domain_conf"].(*schema.Set).List()) != 0 {
+			domainConfig, err := expandVsanFileServiceDomainConfig(d, fileServiceConf)
+			if err != nil {
+				return err
+			}
+			if err := vsanclient.CreateFileServiceDomain(meta.(*Client).vsanClient, meta.(*Client).vimClient, domainConfig, cluster.Reference()); err != nil {
+				return fmt.Errorf("cannot configure vsan file service domain, err: %s", err)
+			} else {
+				log.Printf("[DEBUG] configure vsan file service domain")
+			}
+		}
+	}
+
+	if !d.Get("vsan_file_service_enabled").(bool) && d.HasChange("vsan_file_service_enabled") {
```

Comment:
For this comment and the next one, we're planning to combine two fields as former features (fd/sc) did, if the block start to appear it means we're going to enable file service, if the content of block changed, it means reconfigure file service, and if we remove whole block from .tf file, it means disable file service, however in this PR, for the first version of this feature, only basic enable/disable will be supported, since reconfigure fs need new vmodl API, we'll add inline comments accordingly.